### PR TITLE
Use LINE emoji in messages 

### LINKED ISSUE #199: Support LINE emoji in messages
https://developers.line.biz/en/news/2020/04/14/messaging-api-update-april-2020/

### PR DESCRIPTION
**Pull request by kkdai**
Fixed #199 
https://developers.line.biz/en/news/2020/04/14/messaging-api-update-april-2020/

### CODE REVIEW

**Comment by kkdai** [comment]
@clsung  update by your suggestion, thank you

File: linebot/emoji.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+// Emoji type
+type Emoji struct {
+	Index     int
+	ProductID string
+	EmojiID   string
+}
+
+// MarshalJSON method of Emoji
+func (s *Emoji) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
```
Comment (nasa9084) [inline linebot/emoji.go:30]: there's no added/removed fields from the struct to the JSON, I think no need proxy object

File: linebot/message.go [hunk 2]
```diff
@@ -83,6 +86,12 @@ func (m *TextMessage) WithSender(sender *Sender) SendingMessage {
 	return m
 }
 
+// AddEmoji method of TextMessage
```
Comment (nasa9084) [inline linebot/message.go:89]: maybe need to add this method to `SendingMessage` interface?

File: linebot/emoji.go (hunk 1 above)
Comment (kkdai) [reply linebot/emoji.go]: Thank you, fixed already.

File: linebot/message.go (hunk 2 above)
Comment (kkdai) [reply linebot/message.go:89]: @nasa9084 -san,
The emoji value only works for TextMessage, and not working for ImageMessage or other message types.

If I add it as an interface for XendMessage, I might do nothing if message type is not TextMessage.

How do you think?

e.g. 

```
// AddEmoji method of ImageMessage
func (m *ImageMessage) AddEmoji(emoji *Emoji) SendingMessage {
	return m
}
```

Comment (nasa9084) [reply linebot/message.go:89]: I'm newer person for this project so I'm not sure though, I think the method is considered to be used as method chain like `m.AddEmoji(emoji1).AddEmoji(emoji2)`.

I can see:

1. `AddEmoji` is returning `SendingMessage` interface
2. the field `emojis` is private field
3. `NewTextMessage` does not have argument to set emojis

so users cannot set multiple emojis without type assertion, if `SendingMessage` does not have `AddEmoji` method.

I think we have some options:

* Add `AddEmoji` method to `SendingMessage` interface
* Change not to return `SendingMessage` for all methods, means remove `SendingMessage` interface
* Change `AddEmoji` method to `AddEmojis` to be able to pass `[]*Emoji`
* Change the `NewTextMessage` to be able to pass emojis
* ... or maybe other choice

HDYT?

cc/ @oklahomer @k2wanko

Comment (kkdai) [reply linebot/message.go:89]: @nasa9084 @oklahomer  Any conclusion?

Comment (kkdai) [reply linebot/message.go:90]: @nasa9084 Could you check again? I have fixed your question.

Comment (nasa9084) [reply linebot/message.go:90]: sorry to be late reply, please wait for a while....

Comment (nasa9084) [reply linebot/message.go:90]: Thank you, I checked and LGTM

**Comment by nasa9084** [comment]
sorry to be late merging

**Pull request merged**